### ISSUE TITLE
Can't use input or output on xp hopper?

### ISSUE THREAD

**Issue by teamaonn** [Wabbit0101/mods_hoardercraft#76]
So I've tried running hoppers, Transfer pipes, relays, pretty much every vanilla and mod option i have to automate importing and exporting bottles of exp in and out of the XP hopper, but it gets me no where. Was this intentional?

**Comment by Wabbit0101** [comment]
Have you tried the following:
1. Putting empty bottles into the top rightmost slot? You can also put melon slices there to create Big Slices o' Enchanting (for eating directly). Point the xp hopper to a vanilla output inventory (like chest or other hopper...SmartHoppers is not yet updated to use non-inventory pipes etc...coming for 1.14)

If neither of these is working then yea something is broken.
Edit -- you can only feed that top rightmost slot from above (all other sides go in as inputs). To pull out the results you need to pull from bottom of hopper or point xp-hopper to another inventory.

**Comment by teamaonn** [comment]
So, yeah that fixed things coming out, but I still can't feed things into it for some odd reason....

**Comment by teamaonn** [comment]
Okay, i can feed xp bottles in thru the top, and front, but I can't put in the containers. I'm trying to feed  storage jars as the output containers, and the smart hopper likes them as output, but won't take them in so that the hopper can stay constantly filled with them. Does that make any sense?

**Comment by Wabbit0101** [comment]
If you mean the smarthopper's storage jars you don't need to provide those (a new concentrate is made from um magic as more xp comes in...seriously it's like me + 5 other people using this mod so I was never really forced to explain how that magic worked...) 

As for getting concentrate jars out (I'm assuming you want full concentrate jars?)...then put a slot plug in that top rightmost slot. When a concentrate jar is full, it's sealed and moved to an output slot where you can pipe it out the hopper. A new concentrate jar begins filling...without you needing to supply anything.

The concentrate jars are useful for storage. Right-clicking them will level you up quickly also. To transfer your player xp to an xp-hopper, just make one of those xp transfer doodads, hold it and jump on the top of an xp-hopper (yea, it's weird). To transfer your xp directly to bottles o' enchanting, just hold empty bottles in your offhand, and right click the transfer doodad.

HTH

**Comment by teamaonn** [comment]
yeah, big jars won't go into that upper right corner, nor can i use them like the little storage jars. and I'm having a hard time setting those up for an autorefil thing.

**Comment by Wabbit0101** [comment]
Currently those big Xp jars/blocks are there solely for hand-crafting to allow mass storage using vanilla-isque "put stuff in a chest or shulkerbox" when forge fluids aren't present or players aren't that tech-savy. My assumption was for anything more complex for such large amounts of xp, the player would most likely store experience as a liquid in a ginormous tank of some kind associated with their mob farm(s). 

Can you say exactly what it is you're trying to do? (I _think_ you're trying to autocraft the big storage jars and blocks which unfortunately is a recipe to big for the mod's autocrafting hoppers.) Could be an interesting use case for the 1.14 rewrite of smarthoppers.
